whylog/config/__init__.py

```diff
@@ -30,14 +34,21 @@ def _load_rules(self):
     def _load_log_types(self):
         pass
 
+    def _get_parsers_name(self):
+        parsers_name = OrderedSet()
+        for parser in self._parsers:
+            parsers_name.add(parser)
+        return parsers_name
+
     def add_rule(self, user_rule_intent):
         created_rule = RegexRuleFactory.create_from_intent(user_rule_intent)
         self._save_rule_definition(created_rule.serialize())
```

Comment:
What about checking if user_rule_intent has free pattern name? If we have one Teacher this problem doesn't exists. In case of web version of Whylog there will be a problem.